### PR TITLE
Hero Block

### LINKED ISSUE #3: Hero
Three types:

Angled:
Angle

Product:
product

Card Overlay:
Card_overlay

### PR DESCRIPTION
**Pull request by bstopp**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #3 

Test URLs:
- After: https://feat-hero--takeda-ihs--hlxsites.hlx.page/drafts/stopp/hero
- After: https://feat-hero--takeda-ihs--hlxsites.hlx.page/drafts/stopp/hero-emphasis
- After: https://feat-hero--takeda-ihs--hlxsites.hlx.page/drafts/stopp/hero-product
- After: https://feat-hero--takeda-ihs--hlxsites.hlx.page/drafts/stopp/hero-card

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -16,7 +16,7 @@ header {
     / min-content 1fr min-content;
   justify-content: space-between;
   background-color: transparent;
-  z-index: 1000;
```
Comment (lwintch) [inline blocks/header/header.css:19]: thank you

File: blocks/header/header.js [hunk 2]
```diff
@@ -160,7 +160,16 @@ function buildSections(sections) {
           }
         }
       });
-
+      section.addEventListener('pointerleave', (e) => {
+        if (e.pointerType === 'mouse') {
+          toggleAllNavSections(sections);
+          section.setAttribute('aria-expanded', 'false');
```
Comment (lwintch) [inline blocks/header/header.js:166]: Nice touch

File: blocks/hero/hero.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  contentDiv.classList.add('content');
+
+  if (config.title) {
+    const idx = keys.indexOf('title');
+    const content = block.children[idx].children[1];
+    content.classList.add('title');
+    contentDiv.append(content);
+  }
+
+  if (config.resources) {
```
Comment (lwintch) [inline blocks/hero/hero.js:64]: Is there an upper limit that you can think of where the number of elements would negatively impact performance when running this block?

File: blocks/hero/product-hero.css [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  }
+}
+
+@media screen and (min-width: 1200px) {
+  .hero.block.product .content-wrapper {
+    max-width: 980px;
+  }
+}
+
+@media screen and (min-width: 1280px) {
```
Comment (lwintch) [inline blocks/hero/product-hero.css:133]: I noticed a `1280px` breakpoint on the live site when I was working on columns.  Should I go back and refactor to include this new breakpoint?

File: blocks/hero/hero.js (hunk 3 above)
Comment (bstopp) [reply blocks/hero/hero.js:64]: Do you mean number of variations? Not that i can think of, no.

File: styles/styles.css [hunk 5]
```diff
@@ -33,6 +40,15 @@
   --body-font-size-xs: 12px;
   --body-font-size-s: 14px;
   --body-font-size-m: 16px;
+  --body-font-size-l: 18px;
+  --body-font-size-xl: 20px;
+  --body-font-size-xxl: 24px;
+
+  /** Heading **/
+  --heading-font-size-s: 24px;
```
Comment (lwintch) [inline styles/styles.css:48]: Thank you!

**Review by lwintch — APPROVED** [review]
Looks solid.

File: blocks/hero/product-hero.css (hunk 4 above)
Comment (bstopp) [reply blocks/hero/product-hero.css:133]: I am not sure i like this tbh. I don't know about keeping it.

**Comment by bstopp** [comment]
I'm still working on a CLS issue on the Product variation. I don't know what the fix is yet.

File: blocks/hero/product-hero.css (hunk 4 above)
Comment (bstopp) [reply blocks/hero/product-hero.css:133]: I think I need to re-eval the max-widths on these images. The reason there's a 1280 breakpoint has to do with the limited sizes on images at different ports. I'm going to re-assess.

**Comment by bstopp** [comment]
@lwintch - i've done as much as i can do for the CLS, can you please re-review.

File: blocks/hero/hero.js (hunk 3 above)
Comment (lwintch) [reply blocks/hero/hero.js:64]: Cool, thanks!

File: blocks/hero/product-hero.css (hunk 4 above)
Comment (lwintch) [reply blocks/hero/product-hero.css:133]: Awesome.  I'm good with whatever you want to do.  Just let us know when you come to a pattern we should follow.

**Comment by lwintch** [comment]
> I'm still working on a CLS issue on the Product variation. I don't know what the fix is yet. (I should have left this as a draft)

All good.  It happens.

**Review by lwintch — APPROVED** [review]
Nice fix on the nav too!  Tested and looks good.

**Pull request merged**